Code/EditorPlugins/RmlUi/EnginePluginRmlUi/RmlUiAsset/RmlUiView.cpp

```diff
@@ -0,0 +1,65 @@
+#include <EnginePluginRmlUiPCH.h>
+
+#include <Core/ResourceManager/ResourceManager.h>
+#include <EditorEngineProcessFramework/EngineProcess/EngineProcessDocumentContext.h>
+#include <EditorEngineProcessFramework/EngineProcess/EngineProcessMessages.h>
+#include <EnginePluginRmlUi/RmlUiAsset/RmlUiContext.h>
+#include <EnginePluginRmlUi/RmlUiAsset/RmlUiView.h>
+#include <RendererCore/Debug/DebugRenderer.h>
+#include <RendererCore/Pipeline/View.h>
+#include <RendererCore/RenderWorld/RenderWorld.h>
+
+ezRmlUiViewContext::ezRmlUiViewContext(ezRmlUiContext* pRmlUiContext)
+  : ezEngineProcessViewContext(pRmlUiContext)
+{
+  m_pRmlUiContext = pRmlUiContext;
+
+  // Start with something valid.
+  m_Camera.SetCameraMode(ezCameraMode::PerspectiveFixedFovX, 45.0f, 0.01f, 1000.0f);
+  m_Camera.LookAt(ezVec3(1, 1, 1), ezVec3::ZeroVector(), ezVec3(0.0f, 0.0f, 1.0f));
+}
+
+ezRmlUiViewContext::~ezRmlUiViewContext() {}
+
+bool ezRmlUiViewContext::UpdateThumbnailCamera(const ezBoundingBoxSphere& bounds)
+{
+  return !FocusCameraOnObject(m_Camera, bounds, 45.0f, -ezVec3(5, -2, 3));
+}
+
+ezViewHandle ezRmlUiViewContext::CreateView()
+{
+  ezView* pView = nullptr;
+  ezRenderWorld::CreateView("Rml Ui Editor - View", pView);
+
+  pView->SetRenderPipelineResource(CreateDefaultRenderPipeline());
+
+  ezEngineProcessDocumentContext* pDocumentContext = GetDocumentContext();
+  pView->SetWorld(pDocumentContext->GetWorld());
+  pView->SetCamera(&m_Camera);
+  pView->SetCameraUsageHint(ezCameraUsageHint::EditorView);
+  return pView->GetHandle();
+}
+
+void ezRmlUiViewContext::SetCamera(const ezViewRedrawMsgToEngine* pMsg)
+{
+  ezEngineProcessViewContext::SetCamera(pMsg);
```

Comment:
So, do thumbnails + preview work, or not ?

Reply:
Thumbnails and preview work